### PR TITLE
Script for the Mclock testcase#83574902

### PR DESCRIPTION
**Pull request by SrinivasaBharath**
# Description

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

**Comment by SrinivasaBharath** [comment]
Log file-  http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-7QJY0N

Test failed because of - https://bugzilla.redhat.com/show_bug.cgi?id=2124137

File: ceph/rados/core_workflows.py [hunk 1]
```diff
@@ -1033,3 +1034,28 @@ def get_journalctl_log(
         except Exception as er:
             log.error(f"Exception hit while command execution. {er}")
         return log_lines
+
+    def config_param_chk(self, feature):
```
Comment (sunilkumarn417) [inline ceph/rados/core_workflows.py:1038]: can we replace `feature` with `param`?

File: suites/quincy/rados/tier-2_rados_test_mclock_scheduler.yaml [hunk 2]
```diff
@@ -0,0 +1,103 @@
+# Suite contains basic tier-2 rados tests
```
Comment (sunilkumarn417) [inline suites/quincy/rados/tier-2_rados_test_mclock_scheduler.yaml:1]: please plan for complete Mclock test cases like E2E scenario.

File: tests/rados/mclock_test.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+import time
+from configparser import ConfigParser
+
+from ceph.ceph_admin import CephAdmin
+from ceph.rados.core_workflows import RadosOrchestrator
+from utility.log import Log
+
+log = Log(__name__)
+get_config_cmd = "ceph config get osd  "
```
Comment (sunilkumarn417) [inline tests/rados/mclock_test.py:14]: this could be a single definition.

File: tests/rados/mclock_test.py [hunk 4]
```diff
@@ -0,0 +1,107 @@
+"""
```
Comment (sunilkumarn417) [inline tests/rados/mclock_test.py:1]: please add more details into doc string.

File: tests/rados/mclock_test.py [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+    rados_obj = RadosOrchestrator(node=cephadm)
+    cmd = "ceph config get osd osd_mclock_profile"
+    mclock_profiles = ["high_client_ops", "balanced", "high_recovery_ops", "custom"]
+
+    # check for the default mclock profile
+    status = cephadm.shell([cmd])
+    if status[0].strip() != "high_client_ops":
+        log.error("The default Mclock profile is not high_client_ops")
+        return 1
```
Comment (sunilkumarn417) [inline tests/rados/mclock_test.py:31]: I think this could be a validation check.

File: tests/rados/mclock_test.py [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+    try:
+        for profile in mclock_profiles:
+            set_cmd = f"{set_config_cmd} {'osd_mclock_profile'}  {profile}"
+            log.info(f" Changing the Mclock profile to  {profile} profile")
+            rados_obj.run_ceph_command(cmd=set_cmd)
+            time.sleep(3)
+            test_status = config_change_check(rados_obj, profile)
```
Comment (sunilkumarn417) [inline tests/rados/mclock_test.py:38]: please have definition/method to enable profiles.

File: tests/rados/mclock_test.py [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+                return 1
+        return 0
+    except Exception as err:
+        log.error(
+            f" Failed to set or get configuration parameters of the cluster for the  {profile} profile"
+        )
+        log.error(err)
+        return 1
+    finally:
+        set_default_cmd = f"{set_config_cmd} {'osd_mclock_profile high_client_ops'}"
```
Comment (sunilkumarn417) [inline tests/rados/mclock_test.py:52]: why we have set to default profile here?

File: tests/rados/mclock_test.py (hunk 6 above)
Comment (sunilkumarn417) [reply tests/rados/mclock_test.py:38]: get the details from test config.

File: tests/rados/mclock_test.py [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+            get_cmd = f"{get_config_cmd}{param[0]}"
+            status = rados_obj.run_ceph_command(cmd=get_cmd)
+            if profile != "custom":
+                if str(status).strip() != str(param[-1]).strip():
+                    log.error(
+                        f" Able to modify the configuration parameters for the {profile} profile"
+                    )
+                    return 1
+            else:
+                if str(status).strip() == str(param[-1]).strip():
```
Comment (sunilkumarn417) [inline tests/rados/mclock_test.py:82]: please have this comparison in a definition/method.

File: tests/rados/mclock_test.py [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+        log.error(err)
+        return 1
+    finally:
+        set_default_config(rados_obj)
+
+
+def set_default_config(rados_obj: RadosOrchestrator):
+    log.info(" Setting the all Mclock configration parameters to the default values")
+    # Read Config paramters
+    config_info = ConfigParser()
```
Comment (sunilkumarn417) [inline tests/rados/mclock_test.py:99]: I think this could be initialized once at module.

File: tests/rados/rados_config_parameters.ini [hunk 10]
```diff
@@ -0,0 +1,12 @@
...
+   osd_recovery_max_active = 0
+   osd_mclock_scheduler_client_res = 1
+   osd_mclock_scheduler_client_wgt = 1
+   osd_mclock_scheduler_client_lim = 999999
+   osd_mclock_scheduler_background_recovery_res = 1
+   osd_mclock_scheduler_background_recovery_wgt = 1
+   osd_mclock_scheduler_background_recovery_lim = 999999
+   osd_mclock_scheduler_background_best_effort_res = 1
+   osd_mclock_scheduler_background_best_effort_wgt = 1
+   osd_mclock_scheduler_background_best_effort_lim = 999999
```
Comment (sunilkumarn417) [inline tests/rados/rados_config_parameters.ini:12]: please add newline at EOF.

File: ceph/rados/core_workflows.py (hunk 1 above)
Comment (sunilkumarn417) [reply ceph/rados/core_workflows.py:1038]: Update the docstrings with new items been added to workflows.

File: suites/quincy/rados/tier-2_rados_test_mclock_scheduler.yaml [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+      desc: Configures client admin node on cluster
+      module: test_client.py
+      polarion-id:
+      config:
+        command: add
+        id: client.1                      # client Id (<type>.<Id>)
+        node: node7                       # client node
+        install_packages:
+          - ceph-common
+          - ceph-base
```
Comment (pdhiran) [inline suites/quincy/rados/tier-2_rados_test_mclock_scheduler.yaml:92]: Do we need ceph-base package for any client operations in the test?

File: suites/quincy/rados/tier-2_rados_test_mclock_scheduler.yaml (hunk 2 above)
Comment (SrinivasaBharath) [reply suites/quincy/rados/tier-2_rados_test_mclock_scheduler.yaml:1]: A separate PR raised to track End to end scenario.
https://github.com/red-hat-storage/cephci/pull/2062

**Pull request closed without merging**